sources/views/contacts/formContact.js

```diff
@@ -0,0 +1,210 @@
+											rows: [
+												{view: "text", label: "Email", name: "Email"},
+												{view: "text", label: "Skype", name: "Skype"},
+												{view: "text", label: "Phone", name: "Phone"},
+												{
+													view: "datepicker",
+													label: "Birthday",
+													name: "Birthday",
+													format: webix.i18n.longDateFormatStr
+												},
+												{
+													cols: [
+														{
+															template: contact => `<img class='imagePreview' src=${contact.Photo || avatar} />`,
+															name: "Photo",
+															localId: "templateImage",
+															borderless: true,
+															height: 150
+														},
+														{
+															rows: [
+																{},
+																{
+																	view: "uploader",
+																	value: "Change photo",
+																	accept: "image/png, image/gif, image/jpg, image/jpeg",
+																	localId: "imageUploader",
+																	link: "userImage",
+																	autosend: false,
+																	multiple: false
+																},
+																{
+																	view: "button",
+																	value: "Delete Photo",
+																	click: () => this.deletePhoto()
+																}
+															]
+														}
+
+													]
+												}
+											]
+										}
+									]
+								},
+								{},
+								{
+									cols: [
+										{},
+										{
+											view: "button",
+											value: "Cancel",
+											width: 150,
+											click: () => this.closeForm()
+										},
+										{
+											view: "button",
+											localId: "btnAdd",
+											value: "#nameButton#",
+											width: 150,
+											click: () => this.addNewContactOrUpdate()
+										}
+									]
+								}
+							]
+						}
+					],
+					rules: {
+						FirstName: webix.rules.isNotEmpty,
+						LastName: webix.rules.isNotEmpty,
+						StatusID: webix.rules.isNotEmpty,
+						Company: webix.rules.isNotEmpty,
+						Job: webix.rules.isNotEmpty
+					}
+				}
+			]
+		};
+	}
+
+	init() {
+		this.formComponent = this.$$("formForContactData");
+
+		this.image = this.$$("templateImage");
+		const reader = new FileReader();
+		this.$$("imageUploader").attachEvent("onBeforeFileAdd", (upload) => {
+			reader.readAsDataURL(upload.file);
+			reader.onload = () => this.image.setValues({Photo: reader.result});
+		});
+
+		const id = ItemDataContact.getId();
+		const nameForm = id ? "Edit" : "Add";
+		const nameButton = id ? "Save" : "Add";
+		this.$$("header").setValues({nameForm});
+		this.$$("btnAdd").setValue(nameButton);
+		if (id) {
+			let item = ItemDataContact.getItem();
+			this.formComponent.setValues(item);
+			this.image.setValues({Photo: item.Photo});
+		}
+	}
+
+	closeForm(id) {
+		this.formComponent.clear();
+		this.formComponent.clearValidation();
+		ItemDataContact.setId(null);
+		if (id) {
+			this.show(`/top/contacts?id=${id}/contacts.contactDescription`);
```

Comment:
Why did you still using url for transfer id between pages? Use models like ItemDataContact